public/js/displayApplicants.js

```diff
@@ -1,24 +1,77 @@
+  );
+  const select = document.querySelector('#cohorts');
+
+  select.addEventListener('change', () => {
+    // this feels like a hack but best solution I can come up with without rewritting the filter
+    let sort = form.querySelector('.arrowBtn.active').value;
+    let filter = `<input type="hidden" name="sort" value="${sort}">`;
+    form.insertAdjacentHTML('beforeend', filter);
+    form.submit();
+  });
+
+  if (cohort !== 'all') {
+    document
+      .querySelector(`[value='${cohort}']`)
+      .setAttribute('selected', 'true');
+    document
+      .querySelector('.cohortSort')
+      .classList.add('hidden');
+  } else {
+    document
+      .querySelector("[value='all']")
+      .setAttribute('selected', 'true');
+    document
+      .querySelector('.cohortSort')
+      .classList.remove('hidden');
+  }
+})();
+
+//display applicant stages dropdown
+
 (async () => {
-    await outputCohorts();
-
-    const params = new URLSearchParams(window.location.search);
-    const cohort = params.get("cohortId") || 'all';
-    const form = document.querySelector("#sortFilterApplicants");
-    const select = document.querySelector("#cohorts");
-
-    select.addEventListener("change", () => {
-        // this feels like a hack but best solution I can come up with without rewritting the filter
-        let sort = form.querySelector('.arrowBtn.active').value
-        let filter = `<input type="hidden" name="sort" value="${sort}">`
-        form.insertAdjacentHTML('beforeend', filter);
-        form.submit();
-    });
-
-    if (cohort !== 'all') {
-        document.querySelector(`[value='${cohort}']`).setAttribute("selected", "true");
-        document.querySelector('.cohortSort').classList.add('hidden');
-    } else {
-        document.querySelector("[value='all']").setAttribute("selected", "true");
-        document.querySelector('.cohortSort').classList.remove('hidden');
-    }
-})();
\ No newline at end of file
+  await outputStages();
+
+  const params = new URLSearchParams(
+    window.location.search
+  );
+  const stage = params.get('stageId') || 'all';
+  const form = document.querySelector(
+    '#sortFilterApplicants'
+  );
+  const select = document.querySelector('#stages');
+
+  select.addEventListener('change', () => {
+    // this feels like a hack but best solution I can come up with without rewritting the filter
+    let sort = form.querySelector('.arrowBtn.active').value;
+    let filter = `<input type="hidden" name="sort" value="${sort}">`;
+    form.insertAdjacentHTML('beforeend', filter);
+    form.submit();
+  });
+
+  if (stage !== 'all') {
+    document
+      .querySelector(`[value='${stage}']`)
+      .setAttribute('selected', 'true');
+    document
+      .querySelector('.stagesSort')
+      .classList.add('hidden');
+  } else {
+    document
+      .querySelector("[value='all']")
+      .setAttribute('selected', 'true');
+    document
+      .querySelector('.stagesSort')
+      .classList.remove('hidden');
+  }
```

Comment:
Is this doing anything? It looks to me like it's used for the cohorts, because as well as filtering by cohort you can also sort by cohort, but this is not the case for stages